src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala

```diff
@@ -39,13 +39,53 @@ case class PostAssetsItem(
     metadata: Option[Map[String, String]]
 )
 
-class AssetsRelation(config: RelationConfig, assetPath: Option[String])(val sqlContext: SQLContext)
+case class UpdateAssetsItemBase(
+    id: Option[Long],
+    name: Option[String],
+    description: Option[String],
+    metadata: Option[Map[String, String]],
+    source: Option[String],
+    sourceId: Option[String]
+)
+
+case class UpdateAssetsItem(
+    id: Option[Long],
+    name: Option[NonNullableSetter[String]],
+    description: Option[Setter[String]],
+    metadata: Option[Setter[Map[String, String]]],
+    source: Option[Setter[String]],
+    sourceId: Option[Setter[String]]
+)
+object UpdateAssetsItem {
+  def apply(assetItem: UpdateAssetsItemBase): UpdateAssetsItem =
+    new UpdateAssetsItem(
+      assetItem.id,
+      assetItem.name.map(NonNullableSetter(_)),
+      Setter[String](assetItem.description),
+      Setter[Map[String, String]](assetItem.metadata),
+      Setter[String](assetItem.source),
+      Setter[String](assetItem.sourceId)
+    )
+}
+
+class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
     extends CdpRelation[AssetsItem](config, "assets")
     with InsertableRelation
     with CdpConnector {
   @transient lazy private val assetsCreated =
     MetricsSource.getOrCreateCounter(config.metricsPrefix, s"assets.created")
 
+  override def update(rows: Seq[Row]): IO[Unit] = {
+    val assetsItems = rows.map(r => fromRow[UpdateAssetsItemBase](r))
```

Comment:
I think it's readable enough in one line, `val updateAssetsItems = rows.map(r => UpdateAssetsItem(fromRow[UpdateAssetsItemBase](r)))`.

Note "items" in the val name, since we have many. I assume there's a limit on how many we can update in one go, so this should probably be grouped and updated in batches with a `parTraverse`?

Reply:
Agreed, I'll change this elsewhere as well.
This is actually already being grouped in DefaultSource, perhaps we could find a way to do this for  insertInto writing as well.